analyzers/tests/SonarAnalyzer.TestFramework/Common/SourceGeneratorProvider.cs

```diff
@@ -30,25 +30,28 @@ public static class SourceGeneratorProvider
 
     public static AnalyzerFileReference[] SourceGenerators { get; } =
     [
-        new(CheckRazorSourceGeneratorPath(), new AssemblyLoader())
+        new(CheckAndReturnRazorSourceGeneratorPath(), new AssemblyLoader())
     ];
 
-    public static string CheckRazorSourceGeneratorPath() =>
+    public static string CheckAndReturnRazorSourceGeneratorPath() =>
         File.Exists(RazorSourceGeneratorPath) ? RazorSourceGeneratorPath : throw new FileNotFoundException($"Razor sourcegenerator not found: {RazorSourceGeneratorPath}");
 
     public static string LatestSdkFolder()
     {
         var objectAssembly = typeof(object).Assembly;
-        var objectAssemblyDirectory = new FileInfo(objectAssembly.Location).Directory; // C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.4
+        var objectAssemblyDirectory = Directory.GetParent(objectAssembly.Location);    // C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.4
         var dotnetDirectory = objectAssemblyDirectory.Parent.Parent.Parent;            // C:\Program Files\dotnet
         var sdkDirectory = Path.Combine(dotnetDirectory.FullName, "sdk");              // C:\Program Files\dotnet\sdk
+
         if (!Directory.Exists(sdkDirectory))
         {
-            throw new NotSupportedException($"Razor analysis is only supported for .Net Core.");
+            throw new NotSupportedException($"The directory '{sdkDirectory}' does not exist. " +
+                $"This may be because you are not using .NET Core. " +
+                $"Please note that Razor analysis is only supported when using .NET Core.");
         }
-        // List of all sdk directories for the major version
-        var latestSdkMajorDirectories = Directory.GetDirectories(sdkDirectory, $"{objectAssembly.GetName().Version.Major}.*", SearchOption.TopDirectoryOnly);
-        return latestSdkMajorDirectories.OrderByDescending(x => new DirectoryInfo(x).Name).FirstOrDefault(); // Get the latest sdk directory
+        return Directory.GetDirectories(sdkDirectory, $"{objectAssembly.GetName().Version.Major}.*", SearchOption.TopDirectoryOnly) is { Length: > 0 } specificMajorVersionSdkDirectories
+            ? specificMajorVersionSdkDirectories.OrderByDescending(x => Version.Parse(new DirectoryInfo(x).Name)).First()
+            : throw new DirectoryNotFoundException($"SDK directory not found for version {objectAssembly.GetName().Version.Major}");
```

Comment:
This seems convoluted and the unnecessary `null` check on `specificMajorVersionSdkDirectories` is confusing. What do you think of one of these?

```cs
        return Directory.GetDirectories(sdkDirectory, $"{objectAssembly.GetName().Version.Major}.*", SearchOption.TopDirectoryOnly)
            .OrderByDescending(x => Version.Parse(new DirectoryInfo(x).Name))
            .FirstOrDefault() is { } highestSdkFolder
                ? highestSdkFolder
                : throw new DirectoryNotFoundException($"SDK directory not found for version {objectAssembly.GetName().Version.Major}");
```
```cs
        if (Directory.GetDirectories(sdkDirectory, $"{objectAssembly.GetName().Version.Major}.*", SearchOption.TopDirectoryOnly)
            .OrderByDescending(x => Version.Parse(new DirectoryInfo(x).Name))
            .FirstOrDefault() is not { } highestSdkFolder)
        {
            throw new DirectoryNotFoundException($"SDK directory not found for version {objectAssembly.GetName().Version.Major}");
        }
        return highestSdkFolder;
```

Reply:
Used the last one but inverted to keep a (readable) ternary, let me know what you think

Reply:
I'm not sure what you mean. Your solution is identical to my first proposal (which I'm fine with, of course).